internal/postgres/common.go

```diff
@@ -0,0 +1,149 @@
+// Copyright © 2024 Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S. Department of Energy/National Nuclear
+// Security Administration. All rights in the program are reserved by Triad
+// National Security, LLC, and the U.S. Department of Energy/National Nuclear
+// Security Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide license
+// in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do
+// so.
+
+package postgres
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+)
+
+const (
+	xNameRegex = `^x([0-9]{1,4})c([0-7])(s([0-9]{1,4}))?b([0])(n([0-9]{1,4}))?$`
```

Comment:
Is this used for checking of a valid xname? If so, is it possible to use the [`xname`](https://github.com/Cray-HPE/hms-xname) library instead? I'm not sure if we'd actually want to go this route, but I do find it interesting that this is here.

Reply:
After a closer look, it seems like using that library might not be ideal for this situation.

Reply:
Fixed in 2c58ccf.

Yeah, that makes more sense than using the basic regular expression I had...

Reply:
I think using the library makes more sense until either Ochami gets a fork of the xnames repository or xnames are dropped. Whichever comes first.